src/pages/settings/Profile/CustomStatus/StatusClearAfterPage.js

```diff
@@ -0,0 +1,236 @@
+import withLocalize from '../../../../components/withLocalize';
+import compose from '../../../../libs/compose';
+import DateUtils from '../../../../libs/DateUtils';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps} from '../../../../components/withCurrentUserPersonalDetails';
+import personalDetailsPropType from '../../../personalDetailsPropType';
+
+const defaultProps = {
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+
+const propTypes = {
+    currentUserPersonalDetails: personalDetailsPropType,
+    customStatus: PropTypes.shape({
+        clearAfter: PropTypes.string,
+        customDateTemporary: PropTypes.string,
+    }),
+};
+
+/**
+ * @param {string} data - one of the values from CONST.CUSTOM_STATUS_TYPES or data in format YYYY-MM-DD HH:mm
+ * @returns {string}
+ */
+function getInitialDateBasedFromStatusType(data) {
+    switch (data) {
+        case DateUtils.getEndOfToday():
+            return CONST.CUSTOM_STATUS_TYPES.AFTER_TODAY;
+        case CONST.CUSTOM_STATUS_TYPES.NEVER:
+        case '':
+            return CONST.CUSTOM_STATUS_TYPES.NEVER;
+        case false:
+            return CONST.CUSTOM_STATUS_TYPES.AFTER_TODAY;
+        default:
+            return CONST.CUSTOM_STATUS_TYPES.CUSTOM;
+    }
+}
+
+const useValidateCustomDate = (data) => {
+    const localize = useLocalize();
+    const [customDateError, setCustomDateError] = useState('');
+    const [customTimeError, setCustomTimeError] = useState('');
+
+    const validate = () => {
+        const inputData = moment(data, 'YYYY-MM-DD HH:mm:ss');
+        const currentDate = moment().startOf('day');
+        const currentDateTimePlus2 = moment().add(1, 'minutes');
+
+        // Date validation
+        if (inputData.isBefore(currentDate)) {
+            setCustomDateError(localize.translate('common.error.invalidDateShouldBeFuture'));
+        } else {
+            setCustomDateError('');
+        }
+
+        // Time validation
+        if (inputData.isBefore(currentDateTimePlus2)) {
+            setCustomTimeError(localize.translate('common.error.invalidTimeShouldBeFuture'));
+        } else {
+            setCustomTimeError('');
+        }
+    };
+
+    useEffect(() => {
+        validate();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [data]);
+
+    const triggerValidation = () => {
+        validate();
+    };
+
+    return {customDateError, customTimeError, triggerValidation};
+};
+
+function StatusClearAfterPage({currentUserPersonalDetails, customStatus}) {
+    const localize = useLocalize();
+    const clearAfter = lodashGet(currentUserPersonalDetails, 'status.clearAfter', '');
+    const draftClearAfter = lodashGet(customStatus, 'clearAfter', '');
+    const customDateTemporary = lodashGet(customStatus, 'customDateTemporary', '');
+    const [draftPeriod, setDraftPeriod] = useState(getInitialDateBasedFromStatusType(clearAfter || draftClearAfter));
+    const localesToThemes = useMemo(
```

Comment:
I guess `themes` refers to the preset options (30 Mins, 1 Hour, etc.)? Would be nice to standardise on how it's referred to in other functions. I think "status type" is somewhat more descriptive (and is used more widely in your code), but don't feel too strongly on it, as long as it's consistent!